File: samples/php/channel_start.php
<?php
        /*
         * Start a channel on a queuemanager
         */

        if ( count($argv) < 3 ) {
                echo 'Please pass a queuemanager name and channelname as argument', PHP_EOL;
                exit;
        }
        $qmgr = $argv[1];
        $channel = $argv[2];

        $url = "http://localhost:8081/api/channel/start/$qmgr/$channel";

        $curl = curl_init();
        curl_setopt($curl, CURLOPT_URL, $url);
        curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);

        if ( ($response = curl_exec($curl) ) === false ) {
                $err = curl_error($curl);
                echo "An HTTP error occurred while starting channel $channel: $err\n";
        }
        else {
                $json = json_decode($response);
                if ( isset($json->error) ) {
                        echo "An MQ error occurred while starting channel $channel.\n";
                        echo "Reason Code: {$json->error->reason->code} - {$json->error->reason->desc}\n";
                }
                else {
                        echo "Channel $channel started.\n";
                }
        }

File: samples/python/channel_start.py
'''
 Starts a channel on a queuemanager
 MQWeb runs on localhost and is listening on port 8081.
'''
import json
import httplib
import socket
import argparse

parser = argparse.ArgumentParser(
	description='MQWeb - Python sample - Start Channel',
	epilog="For more information: http://www.mqweb.org"
)
parser.add_argument('-m', '--queuemanager', help='Name of the queuemanager', required=True)
parser.add_argument('-c', '--channel', help='Name of the channel', required=True)
args = parser.parse_args()

url = "/api/channel/start/" + args.queuemanager + '/' + args.channel;

try:
	conn = httplib.HTTPConnection('localhost', 8081)
	conn.request('GET', url)
	res = conn.getresponse()
	result = json.loads(res.read())

	if 'error' in result:
		print ('Received a WebSphere MQ error: ' +
			str(result['error']['reason']['code'])
		)
	else:
		print "Start command succesfully send."
		print "Note: this doesn't mean the channel is running! Use chstatus to get the status of the channel."

except httplib.HTTPException as e:
	print ('An HTTP error occurred while starting a channel: ' +
		e.errno + e.strerror
	)
except socket.error as e:
	print e.strerror
	print 'Is the MQWeb daemon running?'
